### PR TITLE
Milestone 3 | query multiple tables

### PR DESCRIPTION
**Pull request by ITurres**
# 🚩I have made the following changes to complete Vet Clinic - Milestone/Project 3

### Type of **`files`** changed/added

My Skills

### In this new Branch, I have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this milestone's [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md)

# \*📂 root/

- **\*\*📄 data.sql** ✏️

  - I have inserted data into the `owners` and `species` tables
  - I have also updated the `animals` table with the data related to the `owners` and `species` tables.

- **\*\*📄 queries.sql** ✏️

  - New queries have been made to the `animals`, `owners` and `species` tables.

- **\*\*📄 schema.sql** ✏️

  - I have removed the **species** column and added the following:

    - **species_id** column.
    - **owner_id** column.
    - Two **FOREIGN KEY** constraints to the `species` and `owners` tables.

  - I created the two tables mentioned above:

    - `species` table.
    - `owners` table.

---

### 📸 Screenshots of queries made to the database

  <details>
    <summary>SQL Query Collection- Click to expand!</summary>

vet-clinic-project-3-queries-123

vet-clinic-project-3-queries-456

vet-clinic-project-3-queries-7last

  </details>

---

### Thank you for taking the time to review this PR ⭐

---

If you require additional information or have any questions, please contact me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_. I'll be happy to help you out. 🎯

---

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,23 @@ CREATE TABLE
         escape_attempts SMALLINT NOT NULL,
         neutered BOOLEAN NOT NULL,
         weight_kg NUMERIC(10, 2) NOT NULL,
-        species VARCHAR(50) NULL
+        species_id INT,
+        owner_id INT,
+        CONSTRAINT fk_species FOREIGN KEY (species_id) REFERENCES species (id) ON DELETE CASCADE,
+        CONSTRAINT fk_owners FOREIGN KEY (owner_id) REFERENCES owners (id) ON DELETE CASCADE
+    );
```
Comment (bobb-Rob) [inline schema.sql]: - [x] When implementing modifications to existing database tables, it is considered a recommended practice to utilize the ALTER keyword for executing new statements that enact these changes. In this context:

- The "species" column is being removed.
- A new column named "species_id" is being introduced, establishing a foreign key relationship with the "species" table.
- An additional column named "owner_id" is being appended, establishing a foreign key relationship with the "owners" table.

The SQL commands for these actions can be structured as follows:

```sql
ALTER TABLE animals DROP COLUMN species;
ALTER TABLE animals ADD COLUMN species_id INT REFERENCES species(id);
ALTER TABLE animals ADD COLUMN owner_id INT REFERENCES owners(id);
```

This approach ensures that alterations to the database schema are carried out in a controlled manner, aligning with best practices for maintaining data integrity and database structure.

Please remove the commands from lines 9 to 12 for the animal table and use the above-suggested method.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @ITurres, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ Data insertion is perfectly done
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (ITurres) [reply schema.sql]: Hi @bobb-Rob I hope you are well and cheers for reviewing this pr! ⭐️

---

All changes were executed using the **psql** tool:

I utilized the ALTER command to appropriately modify tables for deletion and updating in accordance with the new requirements.

These SQL statements were implemented via **psql** to address the subsequent requirements.

However, the clarity regarding whether these statements should be documented or if direct modifications to the target table were necessary was not provided by the requirements or were ambiguous.

If you require additional details or have any inquiries, please don't hesitate to let me know. 🤠

Comment (Davidosky007) [reply schema.sql]: * As mentioned in the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md#:~:text=Remove%20column%20species) you are to remove the species column, currently, your queries hasn't solved this yet. kindly fix this

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ITurres 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR 👍 
* Good looking readme 👍 
* Multiple tables queried 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (ITurres) [reply schema.sql]: Hi Guys! Good day 🌞

All right then, changes will be made to follow the suggested required changes 🎯 

Cheers again! 😃

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @ITurres , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ITurres** [comment]
Hi @NahnahAJ! thank you for reviewing this PR too! :D I want to thank you all @bobb-Rob && @Davidosky007 for providing insightful feedback and pointing me in the right direction! 🎯 it is always very appreciated.

I wish you all a good week and happy coding! 🪂

**Pull request merged**